dspace-server-webapp/src/main/java/org/dspace/app/rest/authorization/impl/CanCreateSubmission.java

```diff
@@ -0,0 +1,114 @@
+import org.dspace.app.rest.model.CollectionRest;
+import org.dspace.app.rest.model.SiteRest;
+import org.dspace.authorize.service.AuthorizeService;
+import org.dspace.content.Item;
+import org.dspace.content.service.ItemService;
+import org.dspace.core.Context;
+import org.dspace.eperson.EPerson;
+import org.dspace.eperson.Group;
+import org.dspace.services.ConfigurationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Streams;
+
+
+/**
+ * This authorization feature is a bit different from 'canSubmit'.
+ * In this feature we rather check if the user can make many submissions, only one or none.
+ * This is useful in cases where a student can only make one submission (for a master thesis for ex.)
+ * but where an administrator account can submit multiple times.
+ */
+@Component
+@AuthorizationFeatureDocumentation(name = CanCreateSubmission.NAME, 
+    description = "It can be used to verify if a user has the rights to create a new submission in the repository.")
+public class CanCreateSubmission implements AuthorizationFeature{
+    public static final String NAME = "canCreateSubmission";
+
+    @Autowired
+    AuthorizeService authService;
+
+    @Autowired
+    ItemService itemService;
+
+    @Autowired
+    ConfigurationService configurationService;
+
+    @Override
+    public boolean isAuthorized(Context context, BaseObjectRest object) throws SQLException {
+        return this.isUserAuthorized(context);
+    }
+
+    @Override
+    public String[] getSupportedTypes() {
+        return new String[] {
+            CollectionRest.CATEGORY + "." + CollectionRest.NAME,
+            SiteRest.CATEGORY + "." + SiteRest.NAME
+        };
+    }
+
+    /**
+     * Check if the user can make a new submission based on his roles.
+     * 
+     * @param ctx: The current DSpace context.
+     * @return: True if the user can submit, false if not :).
+    */
+    private Boolean isUserAuthorized(Context ctx) throws SQLException {
+        EPerson currentUser = ctx.getCurrentUser();
+        if (currentUser != null) {
+            for (Group group: currentUser.getGroups()){
+                // Two different cases to accept the submission creation:
+                // 1. - The user is an administrator, manager... 
+                // 2. - The user is a student which has no pending submissions
+                if (this.canSubmitMultipleTimes(group.getName()) || (this.numberOfCurrentSubmissionForEperson(ctx, currentUser) < 1)) {
+                    return true;
+                };
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Check if the user can make many submissions.
+     * 
+     * @param searchedValue: A role (a string) of a user.
+     * @return: True if the user can submit multiple times, false if not.
+    */
+    private boolean canSubmitMultipleTimes(String searchedValue) {
```

Comment:
`searchValue` --> `roleName`
In docstring: not needed to specify "(a string)" (it's already into the function signature)